config/v3_0/types/mode_test.go

```diff
@@ -52,11 +53,50 @@ func TestModeValidate(t *testing.T) {
 			errors.ErrFileIllegalMode,
 		},
 	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("validate %d", i), func(t *testing.T) {
+			actual := validateMode(test.in)
+			expected := test.out
+			assert.Equal(t, actual, expected, "bad validation for mode")
+		})
+	}
+}
 
+func TestPermissionBitsValidate(t *testing.T) {
+	tests := []struct {
+		in  *int
+		out error
+	}{
+		{
+			nil,
+			nil,
+		},
+		{
+			util.IntToPtr(0),
+			nil,
+		},
+		{
+			util.IntToPtr(0644),
+			nil,
+		},
+		{
+			util.IntToPtr(0755),
+			nil,
+		},
+		{
+			util.IntToPtr(0777),
+			nil,
+		},
+		{
+			util.IntToPtr(1005),
```

Comment:
Let's write this in octal for clarity.  Would be good to add tests for the other special bits also.